### DISCUSSION TITLE
Provide a React function component "as prop" from a server component to a client component

### DISCUSSION THREAD

**Discussion by lorenzo-dallamuta** [vercel/next.js#71301] [Help]
### Summary

I have a component that follows a bit of an unusual pattern: it receives a list of items and React component (so a function), then internally it iterates on the items and passes each one to an invocation of the provided React component.

In other words, it's a bit like a "templatable" container, which receives a dynamic card component and a list of items to feed as props to the cards.

```
"use client"

function ComponentA(items, SubComponent) {
    return items.map((item) => <SubComponent {...item}>)
}
```

```
"use client"

function ComponentB(item) {
    return // ...does things with item
}
```

```
function ComponentC() {
    return <ComponentA items=[...] SubComponent={ComponentB} />
}
```

in Next this only works if ComponentC itself is a client component, otherwise I get the error `Functions cannot be passed directly to Client Components...`, but in my understanding... the invocation of ComponentB happens inside of ComponentA.

Is it at all possibile to provide a component "as prop" from a server component to a client component, assuming the invocation itself doesn't happen in the server side?

**Comment by icyJoseph** [comment]
Is it possible for your pattern to move the `use client` directive, elsewhere?

**Reply by lorenzo-dallamuta** [reply]
The actual code I'm working on uses a library that requires to be rendered on the client (at least at its current version).
